generated/server/handlers.go

```diff
@@ -57,19 +58,24 @@ func GetBooksHandler(ctx context.Context, w http.ResponseWriter, r *http.Request
 func NewGetBooksInput(r *http.Request) (*models.GetBooksInput, error) {
 	var input models.GetBooksInput
 
+	var err error
+	_ = err
+
 	authorStr := r.URL.Query().Get("author")
-	input.Author = authorStr
+	authorTmp := authorStr
+	input.Author = &authorTmp
+
 	availableStr := r.URL.Query().Get("available")
-	var err error
-	input.Available, err = strconv.ParseBool(availableStr)
-	if err != nil {
-		return nil, err
-	}
+	availableTmp, err := strconv.ParseBool(availableStr)
```

Comment:
when i originally wrote it i thought it made sense to ignore errors if the parameter wasn't required, but now that I think about it, i should error here if they did specify something and it failed